### PR TITLE
Add user facing errors guidelines + don't use debug to display errors

### PR DESCRIPTION
**Pull request by phillipleblanc**
Adds guidelines on how to provide usable error messages to the user.

Refactors all places that use the Debug trait to log an error, to use the Display trait - as per the new guidelines.

### CODE REVIEW

File: crates/runtime/src/datafusion.rs [hunk 1]
```diff
@@ -339,20 +339,20 @@ impl DataFusion {
             let plan = match ctx.state().statement_to_plan(statements[0].clone()).await {
                 Ok(plan) => plan,
                 Err(e) => {
-                    tracing::error!("Unable to create view: {e:?}");
+                    tracing::error!("Unable to create view: {e}");
```
Comment (lukekim) [inline crates/runtime/src/datafusion.rs]: These should probably all be "Failed to ___" not "Unable to ___"

**Review by lukekim — DISMISSED** [review]

File: docs/dev/user_facing_errors.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Guidelines for displaying errors to the user
+
+1. **Specific**: The error message should be specific enough to help the user understand what went wrong and how to fix it.
+1. **Avoid Debugging Information**: The error message should not contain any debugging information. This includes stack traces, Rust debug representations, or any other technical information that the user cannot act on.
+  - It can be helpful for developers to access this, but it should be gated behind a debug mode/log level/log file, etc and not shown by default.
+1. **Internal/Unknown Errors**: If the error is internal to the application or unknown, the error message should be generic and not expose any internal details. The error message should indicate how to report the error to the developers.
+  - We should strive to have as few of these as possible.
```
Comment (phillipleblanc) [inline docs/dev/user_facing_errors.md]:
```suggestion
  - Strive to have as few of these as possible.
```

File: crates/runtime/src/datafusion.rs (hunk 1 above)
Comment (phillipleblanc) [inline crates/runtime/src/datafusion.rs]:
```suggestion
                    tracing::error!("Failed to create view: {e}");
```

File: crates/runtime/src/datafusion.rs [hunk 3]
```diff
@@ -339,20 +339,20 @@ impl DataFusion {
...
                     return;
                 }
             };
 
             let view = match ViewTable::try_new(plan, Some(sql.to_string())) {
                 Ok(view) => view,
                 Err(e) => {
-                    tracing::error!("Unable to create view: {e:?}");
+                    tracing::error!("Unable to create view: {e}");
```
Comment (phillipleblanc) [inline crates/runtime/src/datafusion.rs]:
```suggestion
                    tracing::error!("Failed to create view: {e}");
```

File: crates/runtime/src/datafusion.rs [hunk 4]
```diff
@@ -339,20 +339,20 @@ impl DataFusion {
...
                     return;
                 }
             };
             if let Err(e) = ctx.register_table(table_name.as_str(), Arc::new(view)) {
-                tracing::error!("Unable to create view: {e:?}");
+                tracing::error!("Unable to create view: {e}");
```
Comment (phillipleblanc) [inline crates/runtime/src/datafusion.rs]:
```suggestion
                tracing::error!("Failed to create view: {e}");
```

File: crates/runtime/src/flight/do_put.rs [hunk 5]
```diff
@@ -53,14 +53,14 @@ pub(crate) async fn handle(
...
-            "No publishers registered for path: {path:?}",
+            "No publishers registered for path: {path}",
         )));
     };
     let dataset = Arc::clone(&publishers.0);
     let data_publishers = Arc::clone(&publishers.1);
 
     let schema = try_schema_from_flatbuffer_bytes(&message.data_header)
-        .map_err(|e| Status::internal(format!("Unable to get schema from data header: {e:?}")))?;
+        .map_err(|e| Status::internal(format!("Unable to get schema from data header: {e}")))?;
```
Comment (phillipleblanc) [inline crates/runtime/src/flight/do_put.rs]:
```suggestion
        .map_err(|e| Status::internal(format!("Failed to get schema from data header: {e}")))?;
```

File: crates/runtime/src/flight/do_put.rs [hunk 6]
```diff
@@ -96,7 +96,7 @@ pub(crate) async fn handle(
                     ) {
                         Ok(batches) => batches,
                         Err(e) => {
-                            tracing::error!("Unable to convert flight data to batches: {e:?}");
+                            tracing::error!("Unable to convert flight data to batches: {e}");
```
Comment (phillipleblanc) [inline crates/runtime/src/flight/do_put.rs]:
```suggestion
                            tracing::error!("Failed to convert flight data to batches: {e}");
```

File: crates/runtime/src/flight/do_put.rs [hunk 7]
```diff
@@ -116,7 +116,7 @@ pub(crate) async fn handle(
                         if let Err(err) = publisher
                             .add_data(Arc::clone(&dataset), data_update.clone())
                             .await
-                            .map_err(|e| Status::internal(format!("Unable to add data: {e:?}")))
+                            .map_err(|e| Status::internal(format!("Unable to add data: {e}")))
```
Comment (phillipleblanc) [inline crates/runtime/src/flight/do_put.rs]:
```suggestion
                            .map_err(|e| Status::internal(format!("Failed to add data: {e}")))
```

File: crates/runtime/src/http/v1.rs [hunk 8]
```diff
@@ -659,8 +659,9 @@ pub(crate) mod inference {
                         };
                     }
                     tracing::error!(
-                        "Unable to cast inference result for model {model_name} to Float32Array: {column_data:?}"
+                        "Unable to cast inference result for model {model_name} to Float32Array"
```
Comment (phillipleblanc) [inline crates/runtime/src/http/v1.rs]:
```suggestion
                        "Failed to cast inference result for model {model_name} to Float32Array"
```

File: crates/runtime/src/http/v1.rs [hunk 9]
```diff
@@ -659,8 +659,9 @@ pub(crate) mod inference {
...
                     );
+                    tracing::debug!("Unable to cast inference result for model {model_name} to Float32Array: {column_data:?}");
```
Comment (phillipleblanc) [inline crates/runtime/src/http/v1.rs]:
```suggestion
                    tracing::debug!("Failed to cast inference result for model {model_name} to Float32Array: {column_data:?}");
```

File: crates/runtime/src/lib.rs [hunk 10]
```diff
@@ -169,7 +169,7 @@ impl Runtime {
                             metrics::counter!("datasets_load_error").increment(1);
                             warn_spaced!(
                                 spaced_tracer,
-                                "Unable to get data connector from source for dataset {}, retrying: {err:?}",
+                                "Unable to get data connector from source for dataset {}, retrying: {err}",
```
Comment (phillipleblanc) [inline crates/runtime/src/lib.rs]:
```suggestion
                                "Failed to get data connector from source for dataset {}, retrying: {err}",
```

File: crates/runtime/src/lib.rs [hunk 11]
```diff
@@ -199,7 +199,7 @@ impl Runtime {
                         metrics::counter!("datasets_load_error").increment(1);
                         warn_spaced!(
                             spaced_tracer,
-                            "Unable to initialize data connector for dataset {}, retrying: {err:?}",
+                            "Unable to initialize data connector for dataset {}, retrying: {err}",
```
Comment (phillipleblanc) [inline crates/runtime/src/lib.rs]:
```suggestion
                            "Failed to initialize data connector for dataset {}, retrying: {err}",
```

File: crates/runtime/src/lib.rs [hunk 12]
```diff
@@ -563,7 +563,7 @@ async fn shutdown_signal() {
     let ctrl_c = async {
         let signal_result = signal::ctrl_c().await;
         if let Err(err) = signal_result {
-            tracing::error!("Unable to listen to shutdown signal: {err:?}");
+            tracing::error!("Unable to listen to shutdown signal: {err}");
```
Comment (phillipleblanc) [inline crates/runtime/src/lib.rs]:
```suggestion
            tracing::error!("Failed to listen to shutdown signal: {err}");
```

**Review by mitchdevenport — APPROVED** [review]

**Review by sgrebnov — APPROVED** [review]

**Pull request merged**